backend/app/api/routes/prompts.py

```
@@ -235,31 +240,40 @@ def read_prompts(
+        list[dict]: List of prompts with user settings matching the filters and sorted as specified.
 
     Raises:
         HTTPException: If an error occurs during database query execution.
     """
     try:
-        # 构建基础查询
+        # -------- Tag IDs preprocessing --------
+        # When this function is invoked internally, the default value of tag_ids can be a
+        # fastapi.params.Query object instead of a real list. To avoid TypeError, we
+        # normalise it to either None or a list of UUIDs.
+        if not tag_ids or not isinstance(tag_ids, (list, tuple, set)):
```

Ruff (0.11.9)

Use `X | Y` in `isinstance` call instead of `(X, Y)`

Convert to `X | Y`

(UP038)